new extension:  roam power previewer

dragonforce2010 opened the pull request:
This extension can preview a website content in the sidedrawer, which gives a better roam experience - neat clean notes and easiy preview the website content without going out of roam, to help you focus more in the roam.

This extension's whole idea is to let you stay in roam to focus on your work without going out and meanwhile keep you roam notes clean, without too much blocks nested within which could disturb a lot.

For now, it can only preview the website content, but in the feature, I tend to have more contents in the roam can be previewed in the same way. Reach me out if you have any feature request.

![](https://firebasestorage.googleapis.com/v0/b/firescript-577a2.appspot.com/o/imgs%2Fapp%2FExploreSpace%2FgnDYv5fQh2.59.25.gif?alt=media&token=5b4399c6-f44f-4da4-a551-dd44b5f7a30f)

panterarocks49 commented on extensions/dragonforce2010/roam-power-previewer.json:
```diff
@@ -0,0 +1,10 @@
+  "short_description": "This extension can preview a website content in the sidedrawer, which gives a better roam experience - neat clean notes and easiy preview the website content without going out of roam, to help you focus more in the roam.",
+  "author": "Michael Zhang",
+  "tags": ["markdown", "sidedrawer", "previewer"],
+  "source_url": "https://github.com/dragonforce2010/roam-power-previewer.git",
```
Can you remove the .git in the url?

dragonforce2010 replied:
sure, will do it, thanks:)

baibhavbista commented:
Hey @dragonforce2010 

- Some comments
    - Reading the README.md, it was not clear to me how to use the extension. Would love if you had GIF or somewhing showing how it works
    - Did you think about making it work for links and not just `[aliased markdown](links)`?
        - When reading the README it was not clear to me that it only worked for aliased links and not normal links (I thought that the extension just plain did not work. I had to go through your code to realize that it worked only for aliased links lol)
    - Maybe it would be possible to capture the error one gets  when website's iframe security policy forbids, and then alert the user to that in the sidedrawer?
        - currently, when it's blocked, user has no way to know. To be precise, they may be confused if the website cannot open or if the website is just taking a long time to load
        
        
- Oh yes one feature request
    - When you said that it was a sidedrawer, I had expected that it would persist even when I minimize it by clicking out of it, and that I could resume it later via something on the side of the screen. Did you consider doing something like that?

baibhavbista reviewed (changes requested):
https://github.com/Roam-Research/roam-depot/pull/544#issuecomment-1498748567

dragonforce2010 commented:
> 

I have addressed your comments, update as below
1. support both pure link and markdown link now
2. support iframe policy check, if target website not support iframe policy, will give the user error prompt
3. enhanced the previewer component
- support window side settings(full screen, 80% and 50%)
- support link copy
- support link title parser
- support close button
- will support more advanced settings through the side drawer title bar button area
4. updated the readme
5. as to your feature request, let me think about it in the next version, this time, I want to release a mvp first

dragonforce2010 commented:
> Hey @dragonforce2010 , requesting a change. please checkout loom https://www.loom.com/share/116feaecc0024b5a8c8122ffe328814b

comments have been addressed, please checkout and review, thanks a lot.

baibhavbista commented:
@dragonforce2010  sorry but here's another review/loom
https://www.loom.com/share/eb2eee09e1e940f48be04e32ed8e42ab

baibhavbista reviewed (changes requested):
@dragonforce2010  sorry but here's another review/loom

https://www.loom.com/share/eb2eee09e1e940f48be04e32ed8e42ab

baibhavbista commented:
thanks for the fixes @dragonforce2010 !
